File: Java/src/com/leetcode/Medium/FindBottomLeftTreeValue.java
package com.leetcode.Medium;

import java.util.*;

// 513. Find Bottom Left Tree Value
public class FindBottomLeftTreeValue {
    public static class TreeNode {
        int val;
        TreeNode left, right;
    }

    public int findBottomLeftValue(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while (!queue.isEmpty()) {
            root = queue.poll();
            if (root.right != null)
                queue.offer(root.right);
            if (root.left != null)
                queue.offer(root.left);
        }

        return root.val;
    }
}

File: JavaScript/Medium/FindBottomLeftTreeValue.js
/**
 * 513. Find Bottom Left Tree Value
 * @param {TreeNode} root
 * @return {number}
 */
const findBottomLeftValue = function (root) {
  let queue = [];
  queue.push(root);
  while (queue.length > 0) {
    root = queue.shift();
    if (root.right !== null)
      queue.push(root.right);
    if (root.left !== null)
      queue.push(root.left);
  }

  return root.val;
};